dpos/state/checkpoint.go

```diff
@@ -472,6 +493,111 @@ func NewRewardData() *RewardData {
 	}
 }
 
+func (c *CheckPoint) Serialize(w io.Writer) (err error) {
+	if err = common.WriteUint32(w, c.Height); err != nil {
+		return
+	}
+
+	if err = common.WriteUint32(w, uint32(c.DutyIndex)); err != nil {
+		return
+	}
+
+	if err = c.persistBytesArray(w, c.CurrentArbitrators); err != nil {
+		return
+	}
+
+	if err = c.persistBytesArray(w, c.CurrentCandidates); err != nil {
+		return
+	}
+
+	if err = c.persistBytesArray(w, c.NextArbitrators); err != nil {
+		return
+	}
+
+	if err = c.persistBytesArray(w, c.NextCandidates); err != nil {
+		return
+	}
+
+	if err = c.CurrentReward.Serialize(w); err != nil {
+		return
+	}
+
+	if err = c.NextReward.Serialize(w); err != nil {
+		return
+	}
+	return c.StateKeyFrame.Serialize(w)
+}
+
+func (c *CheckPoint) Deserialize(r io.Reader) (err error) {
+	if c.Height, err = common.ReadUint32(r); err != nil {
+		return
+	}
+
+	var index uint32
+	if index, err = common.ReadUint32(r); err != nil {
+		return
+	}
+	c.DutyIndex = int(index)
+
+	if c.CurrentArbitrators, err = c.getBytesArray(r); err != nil {
+		return
+	}
+
+	if c.CurrentCandidates, err = c.getBytesArray(r); err != nil {
+		return
+	}
+
+	if c.NextArbitrators, err = c.getBytesArray(r); err != nil {
+		return
+	}
+
+	if c.NextCandidates, err = c.getBytesArray(r); err != nil {
+		return
+	}
+
+	if err = c.CurrentReward.Deserialize(r); err != nil {
+		return
+	}
+
+	if err = c.NextReward.Deserialize(r); err != nil {
+		return
+	}
+
+	return c.StateKeyFrame.Deserialize(r)
+}
+
+func (c *CheckPoint) persistBytesArray(value io.Writer,
```

Comment:
Is it better to call it writeBytesArray ?

Reply:
Ok, fixed it